File: PYTHON/Clase9-Vector/Practica5.py
class VectorInt:
    def __init__(self, n):
        self.n = n
        self.data = [0] * n

    def suma(self):
        s = 0
        for x in self.data:
            s += x
        return s

    def promedio(self):
        if self.n == 0:
            return 0.0
        return self.suma() / self.n

    def imprimir(self):
        print(self.data)


v = VectorInt(5)
v.data[0] = 10
v.data[1] = 20
v.data[2] = 30
v.data[3] = 40
v.data[4] = 50

v.imprimir()
print("Suma =", v.suma())
print("Promedio =", v.promedio())

File: C++/Clase9-Vector/Practica5.cpp
#include <iostream>
using namespace std;

class VectorInt
{
public:
    static const int N = 5;
    int data[N];

    void cargarEjemplo()
    {
        data[0] = 10;
        data[1] = 20;
        data[2] = 30;
        data[3] = 40;
        data[4] = 50;
    }

    int suma()
    {
        int s = 0;
        for (int i = 0; i < N; i++)
            s += data[i];
        return s;
    }

    double promedio()
    {
        return (double)suma() / N;
    }

    void imprimir()
    {
        for (int i = 0; i < N; i++)
        {
            cout << data[i] << " ";
        }
        cout << "\n";
    }
};

int main()
{
    VectorInt v;
    v.cargarEjemplo();

    v.imprimir();
    cout << "Suma = " << v.suma() << "\n";
    cout << "Promedio = " << v.promedio() << "\n";
    return 0;
}